src/jit/morph.cpp

```diff
@@ -3187,24 +3187,35 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
         CORINFO_CLASS_HANDLE objClass = NO_CLASS_HANDLE;
         if (isStructArg)
         {
-            // TODO-1stClassStructs: An OBJ node should not be required for lclVars.
-            if (!actualArg->OperIs(GT_OBJ, GT_MKREFANY))
+            objClass = gtGetStructHandle(argx);
+            if (argx->TypeGet() == TYP_STRUCT)
             {
-                BADCODE("illegal argument tree in fgInitArgInfo");
-            }
-            if (actualArg->OperIs(GT_OBJ))
-            {
-                structSize = actualArg->AsObj()->gtBlkSize;
-                objClass   = actualArg->AsObj()->gtClass;
-                assert(structSize == info.compCompHnd->getClassSize(objClass));
+                if (!actualArg->OperIs(GT_OBJ, GT_MKREFANY, GT_LCL_VAR))
+                {
+                    BADCODE("illegal argument tree in fgInitArgInfo");
+                }
+                // For TYP_STRUCT arguments we must have an OBJ, LCL_VAR or MKREFANY
+                switch (argx->OperGet())
```

Comment:
Shouldn't this be using `actualArg` instead of `argx`? Supposedly `argx` could be a COMMA...

Reply:
Yes, I believe you're right - thanks!